test/unit/concerns/hostgroup_extensions_test.rb

```diff
@@ -0,0 +1,34 @@
+require 'test_plugin_helper'
+
+class HostgroupExtensionsTest < ActiveSupport::TestCase
+  before do
+    @role1 = FactoryGirl.create(:ansible_role)
+    @role2 = FactoryGirl.create(:ansible_role)
+    @role3 = FactoryGirl.create(:ansible_role)
+
+    @hostgroup_parent = FactoryGirl.create(:hostgroup, :ansible_roles => [@role2])
+    @hostgroup = FactoryGirl.create(:hostgroup, :ansible_roles => [@role1])
```

Comment:
I moved the model code to a concern but the tests differ in setup logic so I left them as they were.